Improved by Xuefei Wang

xvwang412 opened the pull request:
I updated this app.

xvwang412 committed:
Improved by Xuefei Wang

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
 library(shinydashboard)
 library(shinyBS)
 library(shinyjs)
```
This package can be removed.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
+## App Meta Data----------------------------------------------------------------
+APP_TITLE  <<- "Title of the app"
+APP_DESCP  <<- paste(
+  "Description of the app",
```
You don't need to say "Description of app"; that is what you're supposed to provide.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
+  skin = "purple",
+  dashboardHeader(title = "Hypothesis Testing", titleWidth = 250,
+                  tags$li(class = "dropdown", actionLink("info", icon("info"))),
+                  tags$li(class = "dropdown",
+                          tags$a(href='https://github.com/EducationShinyAppTeam/BOAST',
+                                 icon("github"))),
```
You may remove these lines.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
 library(shinyBS)
 library(shinyjs)
+library(DT)
+
```
Add library(boastUtils)

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
+      menuItem("Overview", tabName = "Overview", icon = icon("dashboard")),
+      menuItem("Explore", tabName = "Explore", icon = icon("wpexplorer")),
+      menuItem("Challenge", tabName = "Challenge", icon = icon("cogs")),
+      menuItem("Data", tabName = "Data", icon = icon("table")),
```
I do not understand what the purpose of displaying this data table is. We need to identify and articulate the purpose of displaying the data table. If we can't/don't, then we need to remove this tab.

If the purpose is to make the data available, then we can link the user to where they can get the data.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
-        ),
+        tabName = "Overview",
+        withMathJax(),
+        h1(strong("Hypothesis Test for Means")),
```
Remove strong from inside h1.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
+        h1(strong("Hypothesis Test for Means")),
+        p("In this app the goal is to learn about the reasoning of a hypothesis test about proportions."),
+        h2("Instructions"),
+        tags$li("In Part 1 you will look at how the population distribution of all the players' free throw percentages is affected by filtering (restricting attention to a subpopulation)."),
```
Break long lines in your code to improve readability. You can put a return in after "free" and before "throw" to move the line down. The app will still display the text coherently but now your code is easier to read.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
-
         br(),
+        h2("Acknowledgements"),
+        p("This app was developed and programmed in 2017 by David Robinson. The hypothesis testing features in part 2 were edited and improved with additional programming in 2018 by Ryan Voyack. The updated version of this app was improved in 2020 by Xuefei Wang."),
```
See prior comment about code readability.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
+        h2("Acknowledgements"),
+        p("This app was developed and programmed in 2017 by David Robinson. The hypothesis testing features in part 2 were edited and improved with additional programming in 2018 by Ryan Voyack. The updated version of this app was improved in 2020 by Xuefei Wang."),
+        br(),
+        img(src = "fthrow2.png", height = 250, width = 650, algin = "middle"),
```
Alt Text is missing; this must be added

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
-      menuItem("Filtering", tabName = "filtering", icon = icon("wpexplorer")),
-      menuItem("Hypothesis Testing", tabName = "hypothesis_testing", icon = icon("cogs")),
-      menuItem("Data", tabName = "data", icon = icon("table"))
+      menuItem("Overview", tabName = "Overview", icon = icon("dashboard")),
```
Please update the icon here to tachometer-alt

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
-        img(src = "fthrow2.png", height = 250, width = 650, algin = "middle")
+        br(),
+        br(),
+        div(class = "updated", "Last Update: 5/13/2020 by XW.")
```
Update the date to when you last made changes.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
             4,
-
+            
             selectInput("filtertype", h2("Select how you would like to filter."), choices = c(GamesPlayed = "games", FreeThrowAttempts = "FTA")),
```
Please bring this into alignment with the Style Guide (explicit arguments). This will hopefully resolve the accessibility error currently being generated (missing label).

Put spaces into the choices GamesPlayed -> "Games Played".

h2 should be removed.

neilhatfield commented on ui.R:
```diff
@@ -2,99 +2,124 @@ library(shiny)
-        ),
+        tabName = "Explore",
+        withMathJax(),
         fluidRow(
```
Let's add some type of h2 text at the top of the page to give a title to this tab.

neilhatfield commented on ui.R:
```diff
@@ -104,41 +129,37 @@ dashboardPage(
+            
             plotOutput("histogramNBA"),
             # Add rollover for Histogram of Free Throw Proportion Plot
             bsPopover(id = "histogramNBA", title = " ", content = "This histogram filters the NBA players based on games played or free throw attempts. Changing the slider will adjust the number of players that fit the selected criteria.", placement = "left", trigger = "hover", options = NULL)
```
Let's move this text out of a popover and as a paragraph element above everything. The main point of the text feels much more like what the user should do on this page.

neilhatfield commented on ui.R:
```diff
@@ -104,41 +129,37 @@ dashboardPage(
+          
           # A box with information to get students thinking and transitioning into part 2
-          box(width = 12, background = "blue", h4("Try to think about what the median and mean of FT% are and what range you might expect most of the players to fall in. "))
+          box(width = 12, background = "purple", h4("Try to think about what the median and mean of FT% are and what range you might expect most of the players to fall in. "))
```
Remove the box. Move the text into the paragraph that will have the popover text. No h4 usage.

neilhatfield commented on ui.R:
```diff
@@ -104,41 +129,37 @@ dashboardPage(
                 value = c(0, 881)
               )
             ),
             img(src = "Giannis.png", height = 219, width = 300, align = "middle")
```
Alt text is missing; must add

neilhatfield commented on ui.R:
```diff
@@ -104,41 +129,37 @@ dashboardPage(
-
+            h2("Histogram"),
+            
             plotOutput("histogramNBA"),
```
See Style Guide for how to add alt text to this plot.

neilhatfield commented on ui.R:
```diff
@@ -104,41 +129,37 @@ dashboardPage(
-        ),
+        tabName = "Challenge",
+        withMathJax(),
         fluidRow(
```
Add an h2 title for this page.

neilhatfield commented on ui.R:
```diff
@@ -104,41 +129,37 @@ dashboardPage(
           column(
             12,
-            h4("Here, we will create a sample 'p-hat' for any player's free throw percentage and test it against a particular null hypothesis. By default, we test it against the NBA average percentage of 74%. Also, we will only be selecting from all NBA players that played no less than half of the games their teams played in during the 2016-2017 season.")
+            h4("Here, we will create a sample 'p-hat' for any player's free throw percentage and test it against a particular null hypothesis. By default, we test it against the NBA average percentage of 74%. Also, we will only be selecting from all NBA players that played no less than half of the games their teams played in during the 2018-2019 season.")
```
No h4 usage. This is a paragraph.

neilhatfield commented on ui.R:
```diff
@@ -151,61 +172,61 @@ dashboardPage(
               "input.howToChooseNBA == 'sel'",
```
Be sure to use explicit arguments for all input options. The sliderInput howToChooseNBA is throwing a missing label error.

neilhatfield commented on ui.R:
```diff
@@ -151,61 +172,61 @@ dashboardPage(
               "input.howToChooseNBA == 'sel'",
               selectizeInput(inputId = "player", "Select your player from the drop down list below:", choices = PlayerNames, multiple = FALSE, options = list(placeholder = "Select Your Player"), selected = NULL)
```
See prior comments about explicit arguments

neilhatfield commented on ui.R:
```diff
@@ -228,23 +249,72 @@ dashboardPage(
+          class = "hangingindent",
+          "Carey, R. (2019). boastUtils: BOAST Utilities. (v0.1.0).
+            [R Package]. Available from
+            https://github.com/EducationShinyAppTeam/boastUtils"
+        ),
+        p(
+          class = "hangingindent",
+          "Chang, W. and Borges Ribeio, B. (2018). shinydashboard: Create
+            dashboards with 'Shiny'. (v0.7.1) [R Package]. Available from
+            https://CRAN.R-project.org/package=shinydashboard"
+        ),
+        p(
+          class = "hangingindent",
+          "Chang, W., Cheng, J., Allaire, J., Xie, Y., and McPherson, J.
+            (2019). shiny: Web application framework for R. (v1.4.0)
+            [R Package]. Available from https://CRAN.R-project.org/package=shiny"
+        ),
+        p(
+          class = "hangingindent",
+          "Hanisch, J. (2016). USA Today Sports"
+        ),
+        p(
+          class = "hangingindent",
+          "Hoop Hounds. Free Throw SHooting Tips. Retrieved from http://www.hoophounds.com/articles/free-throw-shooting-tips/"
+        ),
+        p(
+          class = "hangingindent",
+          "NBA (2019), NBA Advanced Stats, [It contains information about performances of NBA players in 2018-2019].
+            Available at https://stats.nba.com/leaders/?Season=2018-19&SeasonType=Regular%20Season&PerMode=Totals"
+        ),
+        p(
+          class = "hangingindent",
+          "Xie, Y., Cheng, J., Tan, X. (2020).DT: A Wrapper of the JavaScript Library 'DataTables'.(v0.13) 
+          [R Package]. Available from https://CRAN.R-project.org/package=DT"
+        ),
+        p(
+          class = "hangingindent",
+          "Wickham, W. (2016). ggplot2: Elegant graphics for data analysis.
+            [R Package]. Springer-Verlag New York. Available from
+            https://ggplot2.tidyverse.org"
```
Make sure to use the new ASA Style

neilhatfield commented on server.R:
```diff
@@ -7,9 +7,9 @@ library(shiny)
 library(shinyjs)
 library(latex2exp)
 library(boastUtils)
+library(DT)
```
Packages to add: tibble

Packages to remove: magrittr, shinyBS, shinyjs

neilhatfield reviewed (changes requested):
Please tackle the comments I added the ui.R file. Be sure to carry any comments forward (i.e., not just where I made them, but to any subsequent instance where they might apply.)

I would like you to move all plots into ggplot2, rather than base graphics. The text size is too small for the plot in the Challenge tab. Please move text out of h4.

xvwang412 committed:
test

xvwang412 committed:
I updated ui and server. It also needs to investigate on "popover".
In addition, can you please take a look on server, line 463-462? There are two comments left from previous student.

neilhatfield reviewed (changes requested):
Here are some things that need to be addressed:

1. General Comments
    - Work on making your code more reader friendly. One key way you can do this is to avoid long lines. In the RStudio editor, there is a grey vertical line that marks off when a line becomes "long". We generally want our lines to be to the left of this vertical line. A few characters past (1-4) is okay, but anything else should be sent to a new line via a return/carriage break.
2. In the UI,
    - Remove DT from the library calls
    - Fix line 7
    - Line 28: the i button in the header does not do anything. Either you need to program the appropriate information in OR remove the i button from the header.
    - Lines 64-84 must be inside a list environment.
    - Where do the pictures come from? Who took them? We need this information in order to include them in the app.
    - We had mentioned using percentages rather than absolute width/height values on the pictures.
    - If we keep the image on the Overview page, place the image in a div just like you did for the two buttons.
    - Line 107 should be h2, not h1; come up with a better title for the page.
    - There needs to be more text telling the user what to do, what to notice on the Explore page.
    - Line 109; delete; you only need to call `withMathJax()` once here.
    - Within in the `fluidRow` that begins on line 108, use `column` to divide the space.
        - In the first column use `wellPanel` to replace `sidebarPanel` and adjust the code as appropriate
        - In the second column, place the plot. (I think the `mainPanel` might be interfering with your aria-label for the histogram; possibly the popover too.)
    - Place the picture outside of the fluid row; see prior comments about information/sizing for pictures.
    - Lines 172-180: this text does not need to be in a `fluidRow` or a `column`, just a paragraph tag.
        - Replace the text with the following (place an extra backslash (\) before each listed one):
            > In this activity, you'll test any player's free throw percentage against a null hypothesis (provided the player played in at least half of all games during the 2018-2019 season). You will use the player's free throw percentage as the sample proportion, \\(\\hat{p}\\). The default null hypothesis is \\(p_0=0.74\\); however, you can change this through a slider.
    - Move the controls in to a `wellPanel`
    - We mentioned changing the numericInput to a slider for the null hypothesis.
    - Move the two blocks of Challenge text out of the fluid row that contains the controls and the plot.
    - Place them as their own lines underneath the close the `fluidRow` that contains the controls and the plot.
    - See server section for issues with `uiOutput("text#NBA").
    - Make sure references are consistent with ASA Style (see the Style Guide)
3. In the Server
    - When moving to the Explore tab, something is producing the following warning message `Warning: Removed 27 rows containing non-finite values (stat_bin).`. Track down and fix the warning. (The console should be empty.)
    - Add `theme_bw()` to the ggplot calls
    - I think you've changed the plot but I don't have that code here.
    - Replace `h4` in the `output$text1NBA` and `output$text2NBA` with `p`. (The rendered text is not a header.
    - See comment about readable code.
    - Test Headers can be in `h3` rather than `h4`
        - Use "Normal Approximation Hypothesis Test"
        - Use "Exact [Binomial] Hypothesis Test"
    - I see that you defined both the boastPalette and psuPalette, but I don't see where you used them in your code. If you aren't going to use them, you don't need to define them.

ethanwright3 commented:
This is looking really good Xuefei. I just have a couple small notes along with one major but unusual problem that I discovered. First thing is I don't really like your intro sentence of "In this app the goal..." I just don't think its a clear first sentence. I would advice something like "The goal of this app is to understand the reasoning of a hypothesis test about proportions." Additionally I'm not sure why the slider for free throws goes so high to 881 when the max is 715, but it still works. Additionally when the long drop down list to select player occurs it would be cool to have it in alphabetical order or something like that; I know you can search it but still could be good bonus. 

Finally comes in the significant bug I found. When using select player there are a few players who I by luck stumbled upon that cause the app to have a massive amount of errors. 

![error NBA hypothesis test](https://user-images.githubusercontent.com/65522720/86153388-9dffb900-bacf-11ea-844b-7f87affe7971.png)

A players selected cause errors. Those I could find were Dusty Hannahs, Goran Dragic, Justin Patton, John Henson, John Wall and probably many others. Take a look in the data and try to figure out why these players in particular cause errors to occur.

xvwang412 committed:
I have modified the code. It is ready to review.

neilhatfield reviewed (changes requested):
Here are a few things to change:

1. Add the `shinydashboard` package to the ui.R file and remove it from the server.R file
2. Remove `reshape2`, `tibble`, and `DT` from server.R (these don't seem to do anything anymore and didn't cause an error when I turned them off)
3. See my Teams note about fixing the alt text for plots.
4. For the picture on the Overview page you don't need to include "This picture came fromm 'hoophounds. com'" in the alt text. Just describe the image.
5. The two buttons are not aligned with each other. This has to do with how you've used the unordered list environment. The Explore button is inside the environment and thus attends to the left margin of the list; the Challenge button is outside of the environment and thus attends to the left margin of the page. We want these to look similar.
    - The quickest route would be to close the `tags$ul` after the second button's `div`.
6. See emails about filtering/testing.
7. You need to make the font sizes for the plots bigger
    - This would include the titles, the axes, and the labels.
8. Use "widehat" instead of "hat" in the math code

xvwang412 committed:
Finish updating.

neilhatfield reviewed (changes requested):
Here are some things to take care of:

1. Prior Issues
    - Please refer to Issues #1, #2, #3, #4, and #5 
    - For each one, you will need to
        1. Address the issue if you've not already done so
        2. If the issue is already addressed, add "Closes #[\#]" to your next commit comment/push (each on it's own line). This will help us keep track of which issues are still open.
2. Trace and address Issue #7 
3. In the server.R, lines 475-484 can be simplified to the following:
    ```
    if (input$significanceBounds) {
      g + geom_hline(aes(yintercept = lower), color = "black") +
        geom_hline(aes(yintercept = upper), color = "black")
    }
    ```
   - Note that I used camelCase for `signficanceBounds`; the Style Guide does require the use of camelCase.
4. Resolve Issue #8

This app is getting really close to being finished.

xvwang412 committed:
updated

neilhatfield commented:
Resolves #1 
Closes #2 
Resolves #3 
Resolves #4 
Resolves #5 
Resolves #7 
Resolves #8

neilhatfield commented:
@xvwang412 
Would you please look up who owns the copyright on the picture you've included and post that as a reply to this comment? I wasn't quite sure looking at the references.

neilhatfield commented:
> @xvwang412
> Would you please look up who owns the copyright on the picture you've included and post that as a reply to this comment? I wasn't quite sure looking at the references.

Ended up doing Google Image search; it is the Hoop Hounds reference.

xvwang412 committed:
need to pull the latest version

neilhatfield commented:
Closing in favor of #10
